### DISCUSSION TITLE
Device info lost on restart

### DISCUSSION THREAD

**Discussion by chriswood1001** [zwave-js/zwave-js-ui#875] [Q&A]
I had successfully included 50+ nodes where interview completed, manufacturer, product and attributes were all working correctly. After a power cycle of my RPI every one of my battery devices (~30) come back as "Unknown manufacturer 0xXXXX" and their characteristics are lost. Trying to re-interview fails, even when waking up the device.

Should this information not be retained and if so where? I have /store mounted where I can see my <homehex>.json file, node, uses, and settings; however, I don't see manufacture or manufacture ID data in there. Am I doing something wrong?

**Comment by robertsLando** [comment]
@chriswood1001 Could you provide a full zwavejs silly level log file? Also what z2m version are you using?

**Comment by crazypyro136** [comment]
Did this happen to all of your devices below a certain point, all my nodes up to 30 work as normal everything after that I get the same error you are now. Everything was working fine before. 

Would a silly log from me help?

**Reply by chriswood1001** [reply]
After ~48hrs the devices eventually restored themselves and I've not had this issue since, even after restarts. I originally had the problem with 100% of my battery devices, and because I include powered devices first it meant it was the ~2nd half of my list (battery devices) that exhibited the problem.

Fortunately I've been stable for ~2weeks after many restarts and not seen the issue again.

**Comment by robbpope** [comment]
This exact same thing happened to me and I'm struggling to find a solution.  Fully included about 50 nodes, successfully interviewed and entities created.  Suddenly I'm seeing about 3/4 of those devices reverting to Unknown manufacturer 0x031e.  It's very strange as most of the devices exhibiting this glitch/failure are Innoveli LZW31.  There are a few LZW31 devices in my network that aren't exhibiting this behavior so it can't be specific to the type of device.  Attached is the silly level logs for you.  Would love to figure out what's happening with this installation
[zwavejs_355.log](https://github.com/zwave-js/zwavejs2mqtt/files/6239708/zwavejs_355.log)
[zwavejs2mqtt.log](https://github.com/zwave-js/zwavejs2mqtt/files/6239710/zwavejs2mqtt.log)

**Reply by robertsLando** [reply]
@robbpope Did you try to wait some hours to see if the problem fix itself?

**Reply by robbpope** [reply]
yes I did.  This has been an ongoing issue where I pair all of the devices, watch their information populate and set up my behaviors in Node Red.  Then, a week later I will come back to check in on the installation (it's at a business) and half of the nodes have lost their parameters and all of their fields are either blank or showing 0x0(etc).  I went ahead and rebooted, manually initiated a re-interview on all nodes that are displaying the failure mode described above, and waited overnight with "silly" level logging enabled.  I'm attaching these (hopefully) much more complete logs here.  Check them out and let me know what other info I can get you to help me troubleshoot.  Id be happy to allow you to log into the device via NabuCasa if you'd like.
[zwavejs2mqtt.log](https://github.com/zwave-js/zwavejs2mqtt/files/6245085/zwavejs2mqtt.log)
[zwavejs_355.log](https://github.com/zwave-js/zwavejs2mqtt/files/6245086/zwavejs_355.log)

**Comment by crazypyro136** [comment]
I was able to restore about half of my devices by deleting db1d31ed.json yesterday only to check again this morning and find them unknown again. Like robbpope I have a ton of inovelli about half work and half dont among other devices. Let me know if there is anything I can do to provide additional logs\screenshots whatever. Before yesterday I let it sit for 3 or 4 days before I did anything and no change.

[zwavejs_1.log](https://github.com/zwave-js/zwavejs2mqtt/files/6244290/zwavejs_1.log)
[zwavejs2mqtt.log](https://github.com/zwave-js/zwavejs2mqtt/files/6244294/zwavejs2mqtt.log)

**Reply by crazypyro136** [reply]
Just a little more info I am running in docker on an unraid setup using a HUSBZB-1

**Reply by crazypyro136** [reply]
[db1d31ed.json.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6245057/db1d31ed.json.txt)
[nodes.json.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6245058/nodes.json.txt)

**Comment by johntdyer** [comment]
Running into the same issue, had a restart last night and noticed it today that they were all unknown 

Had this on the 0.9.0 addon, and then upgraded to 0.9.1 and still have the issue


[zwavejs2mqtt.log.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6245012/zwavejs2mqtt.log.zip)
[zwavejs_470.log](https://github.com/zwave-js/zwavejs2mqtt/files/6245005/zwavejs_470.log)

**Comment by robertsLando** [comment]
@AlCalzone any clue what could cause this?

**Reply by AlCalzone** [reply]
This is most likely the cache file getting corrupted by a Docker restart which we already have an issue for. I don't know why that happens, but I know it does not without Docker.
Would be interesting to know if the jsonl files are broken too when it happens.

**Reply by crazypyro136** [reply]
Will these help answer that?


[db1d31ed.metadata.jsonl.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6249593/db1d31ed.metadata.jsonl.txt)
[db1d31ed.values.jsonl.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6249594/db1d31ed.values.jsonl.txt)
[db1d31ed.json.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6249595/db1d31ed.json.txt)

**Reply by AlCalzone** [reply]
I won't be able to look at them until Monday. If anyone who experiences this could share their 3 cache files right when it happens, that could help.

**Reply by AlCalzone** [reply]
Those cache files look pretty complete on first glance. If there are problems with specific nodes, I'll need the node IDs.

**Reply by crazypyro136** [reply]
It is every node ID 34 and higher. If I re-interview the node I can get it to come back

**Reply by AlCalzone** [reply]
If you're experiencing the issues with the cache files you posted, there's something weird going on. The info is definitely in there:
```jsonl
{"k":"{\"nodeId\":34,\"commandClass\":114,\"endpoint\":0,\"property\":\"manufacturerId\"}","v":798}
{"k":"{\"nodeId\":34,\"commandClass\":114,\"endpoint\":0,\"property\":\"productType\"}","v":1}
{"k":"{\"nodeId\":34,\"commandClass\":114,\"endpoint\":0,\"property\":\"productId\"}","v":1}
...
{"k":"{\"nodeId\":35,\"commandClass\":114,\"endpoint\":0,\"property\":\"manufacturerId\"}","v":786}
{"k":"{\"nodeId\":35,\"commandClass\":114,\"endpoint\":0,\"property\":\"productType\"}","v":7680}
{"k":"{\"nodeId\":35,\"commandClass\":114,\"endpoint\":0,\"property\":\"productId\"}","v":7680}
...
{"k":"{\"nodeId\":36,\"commandClass\":114,\"endpoint\":0,\"property\":\"manufacturerId\"}","v":543}
{"k":"{\"nodeId\":36,\"commandClass\":114,\"endpoint\":0,\"property\":\"productType\"}","v":3}
{"k":"{\"nodeId\":36,\"commandClass\":114,\"endpoint\":0,\"property\":\"productId\"}","v":136}
...
{"k":"{\"nodeId\":37,\"commandClass\":114,\"endpoint\":0,\"property\":\"manufacturerId\"}","v":345}
{"k":"{\"nodeId\":37,\"commandClass\":114,\"endpoint\":0,\"property\":\"productType\"}","v":2}
{"k":"{\"nodeId\":37,\"commandClass\":114,\"endpoint\":0,\"property\":\"productId\"}","v":81}
... etc.
```

**Reply by crazypyro136** [reply]
this is node 34, the values even show up in the web console, but still says unknown for some reason. All the devices seem to pretty much function other than the locks when they are in an unknown manufacturer state.

![image](https://user-images.githubusercontent.com/26356098/113523765-88905180-956f-11eb-8c3d-094aeff132a5.png)

**Reply by crazypyro136** [reply]
Looks like the locks dont have the data for the manufacturer 

![image](https://user-images.githubusercontent.com/26356098/113523815-dad17280-956f-11eb-8466-be2503f14be7.png)

**Reply by AlCalzone** [reply]
Indeed, Node 68 is missing some info. Which version of `node-zwave-js` are you running?

**Reply by crazypyro136** [reply]
zwavejs2mqtt: 3.0.3
zwave-js: 7.0.1
home id: 3676123629
 home hex:0xdb1d31ed

Although this started on 7.0.0

**Reply by robbpope** [reply]
An even more complete log of my installation.

[zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6260351/zwavejs2mqtt-store.zip)

**Reply by crazypyro136** [reply]
Unraid updated me to the latest version last night and all my nodes after node 34 are dead again here is the latest version I am running:

zwavejs2mqtt: 3.1.0
zwave-js: 7.1.0

**Reply by AlCalzone** [reply]
Any chance you could chime in here with some more info?
https://github.com/zwave-js/zwavejs2mqtt/issues/998

**Comment by mwolter805** [comment]
Had this same issue and the only way to get the nodes functional was to delete all of the zwaveJS2mqtt config files in the shared volume and have it recreate the devices from the controller. Noticed this issue because the Pi completely locked up sometime last night and had to perform a hard reboot. After the reboot, none of the devices were controllable through HA or zwaveJS2mqtt. Tried re-interviewing the nodes but none of them succeeded.  

Unfortunately, I found this topic after completely blowing out the config but I do have a complete backup (attached). Using an Aeotec Z-Stick on a Pi4 with zwaveJS2mqtt in a docker container. Home Assistant is also running in a docker container on a different machine.

One thing I've noticed that has changed is that zwaveJS2mqtt interviews all nodes at once. In the original version I had installed, not sure what version it was, it would only inspect one device at a time. Not sure if this is related but it does appear to put a significant load on the zwave network while inspecting all nodes at once, whereas before it wasn't as noticeable and the inspection was much faster. 
[Archive.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6319539/Archive.zip)

**Comment by mwolter805** [comment]
@AlCalzone Unfortunately this happened again. Rebooted the Pi a couple times for items unrelated to zwave and it happened again. This time I was able to save the debug before deleting the files. Please take a look, the logs are very peculiar. 

Edit: found issue https://github.com/zwave-js/zwavejs2mqtt/issues/998 and will try the updated container. 

[zwavejs2mqtt-store-2.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6338368/zwavejs2mqtt-store-2.zip)

**Reply by AlCalzone** [reply]
> found issue #998 and will try the updated container.

Mind that you should probably re-interview all nodes once you're on the fixed version.

**Reply by mwolter805** [reply]
Thank you for the quick reply. Removed all configuration files and attempting re-interview the nodes but none of them are responding. However, I am receiving sensor updates so it appears communication with the zwave adapter is good. Any ideas what could be the issue?

Edit: Also updated to the latest version container. 

zwavejs2mqtt: 3.4.0
zwave-js: 7.2.4

```
2021-04-19 20:21:29.928 ERROR ZWAVE: Interview of node 2 has failed: Attempt 4/5 failed
2021-04-19 20:21:31.170 ERROR ZWAVE: Interview of node 3 has failed: Attempt 4/5 failed
2021-04-19 20:21:32.412 ERROR ZWAVE: Interview of node 4 has failed: Attempt 4/5 failed
2021-04-19 20:21:33.659 ERROR ZWAVE: Interview of node 5 has failed: Attempt 4/5 failed
2021-04-19 20:21:41.087 ERROR ZWAVE: Interview of node 6 has failed: Attempt 4/5 failed
2021-04-19 20:21:42.316 ERROR ZWAVE: Interview of node 7 has failed: Attempt 4/5 failed
2021-04-19 20:21:43.552 ERROR ZWAVE: Interview of node 8 has failed: Attempt 4/5 failed
2021-04-19 20:21:44.795 ERROR ZWAVE: Interview of node 9 has failed: Attempt 4/5 failed
2021-04-19 20:21:47.278 ERROR ZWAVE: Interview of node 10 has failed: Attempt 4/5 failed
2021-04-19 20:21:48.514 ERROR ZWAVE: Interview of node 11 has failed: Attempt 4/5 failed
2021-04-19 20:21:59.408 ERROR ZWAVE: Interview of node 12 has failed: Attempt 4/5 failed
2021-04-19 20:22:00.647 ERROR ZWAVE: Interview of node 13 has failed: Attempt 4/5 failed
2021-04-19 20:22:01.889 ERROR ZWAVE: Interview of node 14 has failed: Attempt 4/5 failed
2021-04-19 20:22:08.105 ERROR ZWAVE: Interview of node 15 has failed: Attempt 4/5 failed
2021-04-19 20:22:11.825 ERROR ZWAVE: Interview of node 16 has failed: Attempt 4/5 failed
2021-04-19 20:22:13.063 ERROR ZWAVE: Interview of node 2 has failed: Maximum interview attempts reached
2021-04-19 20:22:14.295 ERROR ZWAVE: Interview of node 3 has failed: Maximum interview attempts reached
2021-04-19 20:22:15.542 ERROR ZWAVE: Interview of node 4 has failed: Maximum interview attempts reached
2021-04-19 20:22:16.778 ERROR ZWAVE: Interview of node 5 has failed: Maximum interview attempts reached
2021-04-19 20:22:18.023 ERROR ZWAVE: Interview of node 6 has failed: Maximum interview attempts reached
2021-04-19 20:22:19.256 ERROR ZWAVE: Interview of node 7 has failed: Maximum interview attempts reached
2021-04-19 20:22:20.495 ERROR ZWAVE: Interview of node 8 has failed: Maximum interview attempts reached
2021-04-19 20:22:25.461 ERROR ZWAVE: Interview of node 9 has failed: Maximum interview attempts reached
2021-04-19 20:22:26.700 ERROR ZWAVE: Interview of node 10 has failed: Maximum interview attempts reached
2021-04-19 20:22:27.932 ERROR ZWAVE: Interview of node 11 has failed: Maximum interview attempts reached
2021-04-19 20:22:29.178 ERROR ZWAVE: Interview of node 12 has failed: Maximum interview attempts reached
2021-04-19 20:22:31.665 ERROR ZWAVE: Interview of node 13 has failed: Maximum interview attempts reached
2021-04-19 20:22:32.899 ERROR ZWAVE: Interview of node 14 has failed: Maximum interview attempts reached
2021-04-19 20:22:35.375 ERROR ZWAVE: Interview of node 15 has failed: Maximum interview attempts reached
```

**Reply by AlCalzone** [reply]
Only if you post me a `zwave-js` logfile 🤷🏻‍♂️

**Reply by mwolter805** [reply]
Here you go. This is a restart of the container with logging at silly. Node five is a HomeSeer switched and turned it on and off to show the network is working. 
[zwavejs2mqtt-store-3.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6339276/zwavejs2mqtt-store-3.zip)
![Screen Shot 2021-04-19 at 2 43 03 PM](https://user-images.githubusercontent.com/24851651/115307086-945a4700-a11d-11eb-8d4f-5d7550b1deec.jpg)

**Reply by mwolter805** [reply]
Apologize, bonehead move. Didn't realize zwave was a separate log. Here are the general and zwave logs at silly. Turned on and off node 5 physically at the switch so you can see the communication. 
[zwavejs2mqtt-store-4.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6339477/zwavejs2mqtt-store-4.zip)

**Reply by mwolter805** [reply]
No amount of rebooting, power-cycling, or re-interviewing would bring the devices online. Completely deleted the configuration and reprogrammed, nothing. Even tried downgrading to v3.3.0. As a last-ditch effort, unplugged the Z-Stick while the container was online and plugged it in again, then rebooted. Nodes then started responding.

**Reply by AlCalzone** [reply]
> Failed to send the command after 1 attempts. Transmission queue full

The stick was stuck and needed a power cycle.

> unplugged the Z-Stick while the container was online

For reference, don't do that. That **can** corrupt the stick's memory contents. Best if you shut down the host, wait a few seconds, then pull and replug the stick and boot again.

**Reply by mwolter805** [reply]
HI Al, 
Totally agree. I had tried power cycling the Pi, which cut the power for the Z-Stick but it didn't help. Wasn't looking forward to doing that while the Pi was powered up, but I remembered I did that the first time zwaveJS2mqtt crashed, and as I mentioned, it was a last resort. 

On the other hand, I believe the Z-Stick has an internal battery. Would think this helps prevent internal memory corruption. 

It's almost like something happened to the Z-Stick while zwaveJS2mqtt was crashed. Like the internal buffer filled and couldn't clear itself, even if the Pi was power cycled. Not sure how unplugging the Z-Stick was different but it definitely resolved the issue.  

There was also a very odd issue that while it was malfunctioning, the logs showed identical status messages received 14, 15 or 16 times. In the latest logs attached, at 2021-04-19 22:42:07.679, the switch (Node 5) was physically turned off once and 14 "false" messages were received.

It's working now but this happened twice on 3.3.0. Hopefully, the SIGTERM or SIGINIT change prevents this from returning.

**Reply by AlCalzone** [reply]
> Would think this helps prevent internal memory corruption.

Might be, I'm not sure.

> the switch (Node 5) was physically turned off once and 14 "false" messages were received.

Might be that the node was trying to reach the controller and re-tried a few times too many :D

**Comment by brgerig** [comment]
I'm also having this problem. Not running Docker, but running on an Enterprise Linux 8 VM using the packaged version. Likely happening when the power flickers and the ESX host restarts (I have a UPS, but the batteries died and I haven't replaced them yet).

**Reply by brgerig** [reply]
It's possible that this might be fixed for me—I had been on 4.2.1, but upgraded a couple days ago to 4.5.1. Had another power failure today, and all of my devices were still identified on restart.

**Reply by robertsLando** [reply]
Yep we worked on a fix for this, the bug shouldn't be present anymore

**Comment by arigit** [comment]
Running zwave-js 7.7.5 (z2m 5.0.4), I had an unintended power cycle in the RPi and when the system came up (HA, zwavejs2mqtt, both running on docker) all the battery-powered devices showed up in unknown state and I noticed zwave-js was trying to re-interview them. 

Just migrated to zwave-js and pretty disappointed that I will need to manually wake up each of the sensors (a dozen). Normal container restarts do not exhibit this problem meaning that volume persistence is working ok. It seems the way zwave-js is handling it's cache is a bit weak and it gets corrupted easily. Any known workaround to make z2m more resilient to restarts?

**Reply by robertsLando** [reply]
The problem seems more something docker related then zwave-js related. @AlCalzone is working in many different fixes related to cache files but seems the problem is still there...

**Reply by AlCalzone** [reply]
@arigit The cache files are append-only most of the time. Every now and then, they get compacted which requires re-writing the files. This was always done by writing a separate file, then renaming and finally deleting the original db. However, I've recently learned that Linux might require an `fsync` on the directory itself to make sure the renaming operation is persisted on disk. But realistically this should have only resulted in a corruption when the power-off happens during or shortly after the compaction.
Nonetheless, a fix for that is included in 7.9.0.

That being said - it can also be that your stick is not responsive, which can look a lot like missing cached data.
https://github.com/zwave-js/zwavejs2mqtt/issues/1388#issuecomment-869496033

**Reply by arigit** [reply]
Thanks for the fast response. I will retest this scenario after updating to 7.9. Prior to the migration to zwavejs2mqtt I've never noticed this happening before (with many battery sensors going to an unknown state), with the same stick. 

I did notice something interesting though, while I manually had to wake up all my zooz ZSE40 presence sensors and the aeotec ZWA012 door sensors (after waiting 3-4 hours for sensors to come up on their own, triggering them to no avail), there was one battery sensor that came back up on its own, the aeotec water sensor ZW122, I didn't reset this one nor tried to wake it up). I am not sure if it came back up immediately when the driver restarted or if it took a few hours. Next time this happens I will take a snapshot of state.

**Comment by sstarcher** [comment]
I filled some of my breakers yesterday and hard powered off on accident some of my systems.  HA+zwave2jsmqtt runs on a NUC and all devices showed up as `Unknown manufacturer 0xXXXX`.  I left them alone for about 18 hours to see if they ever become healthy again, but they are still all in an `Unknown manufacturer 0xXXXX` state.  I'm starting to reinterview them now.

**Reply by sstarcher** [reply]
Not docker to just running directly on Mint linux using systemd

**Reply by AlCalzone** [reply]
As always, I need more info to even begin looking at this.
- Which version of zwavejs2mqtt / of the driver?
- How do the driver logs look like after starting in that state?
- How do the cache files look like?

**Reply by shanestarcher-okta** [reply]
I'll try and reproduce the problem and give you the cache and driver logs.
As for the versions 
zwavejs2mqtt 4.5.2
zwave-js 7.7.2

**Reply by AlCalzone** [reply]
> zwave-js 7.7.2

Make sure you update at least to 7.9.0 (z2m 5.1.0). That version contains an important fix (hopefully the last).

**Reply by sstarcher** [reply]
@AlCalzone I upgraded recently to newer version and my last forced power restart did not have the prior issues.

**Comment by mwolter805** [comment]
@AlCalzone Hello, unfortunately, had the same issue appear that was mentioned in the thread https://github.com/zwave-js/zwavejs2mqtt/discussions/875#discussioncomment-636129. All devices in the zwavejs2mqtt web GUI were stuck in Unknown, then ProtocolInfo when trying to reinterview. 

zwavejs2mqtt: 5.4.5
zwave-js: 8.0.8

This occurred after a scheduled reboot of the RPi running the docker container. 

Only way to clear the issue was to unplug the zstick with everything running and reconnect it. 

Below are the console logs from the container. The reboot happened at 09:04:37
[at reboot.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6995798/at.reboot.txt)
This continues on for hours until the zstick is unplugged and reconnected. Alternating between the same messages from node 2 and 3. 

Below are the logs from when the zstick was reconnected
[after zstick reconnection.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6995802/after.zstick.reconnection.txt)
The logs continue to show each node interviewed until it completes. 

Below are the zwavejs logs
[zwavejs_2021-08-16.log.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6995783/zwavejs_2021-08-16.log.zip)

**Reply by AlCalzone** [reply]
@mwolter805 Thanks for the logs. Looks like we need some kind of detection if the Z-Wave stick is stuck and reset it.

**Reply by mwolter805** [reply]
That would be great. 

Not sure what causes it to stop working. Have rebooted the pi dozens of times without issue but this time something went wrong.

I could see this deterring users from the integration as the fix is elusive as nothing other than physically reconnecting the zstick resolves the issue. Tried deleting the config, the entire container, etc. nothing else worked. This has been experienced by other users as I've helped someone with the same issue on the HA forums. Wouldn't be far less frustrating if normal reboot fixed it as I could detect the zwave devices down and reboot the pi with an automation or bash script.
